File: Client/src/app/cards/card.ts
// export class Card {
//     _id: string;
//     idName: string;
//     rarity: string;
//     type: string;
//     name: string;
//     description: string;
//     elixirCost: number;
//     copyId: number;
//     arena: number;
//     order: number;
//     __v: number;
//     imageUrl: string;
// }

export class Card {
    id: number;
    idName: string;
    rarity: string;
    type: string;
    name: string;
    description: string;
    elixirCost: number;
    version: number;
    imageUrl: string;
    targets: string;
    hitSpeed: number;
    count: number;
    range: string
    speed: string;
    radius: number;
    deployTime: number;
    lifetime: number;
    dashRange: string;
    projectileRange: number;
    cardStatistics: CardStatistics[] = [];
}

export class CardStatistics {
    cardLevel: number;
    hitPoints: number;
    damange: number;
    damagePerSecond: number;
    dashDamage: number;
    areaDamage: number;
    crownTowerDamage: number;
    chargeDamage: number;
    shieldHitpoints: number;
    duration: number;
    healingPerSecond: number;
}
export class CardDetail {
    constructor() {
        this.troops = [];
        this.buildings = [];
        this.spells = [];
    }

    troops: Card[];
    buildings: Card[];
    spells: Card[];
}

export class SearchTermCardFilter {
    constructor() {
        this.searchTerm = '';
        this.rarity = 'all';
    }
    searchTerm: string;
    rarity: string;
}

File: ClashRoyaleApi/ClashRoyaleApi.Infrastructure/Models/Card.cs
using ClashRoyaleApi.Infrastructure.Attributes;
using System.Collections.Generic;

namespace ClashRoyaleApi.Infrastructure.Models
{
    public class Card
    {
        [Sortable]
        public int Id { get; set; }

        public string IdName { get; set; }

        [Sortable(IsDefault = true)]
        [SearchableString]
        public string Name { get; set; }

        [Sortable]
        [SearchableString]
        public string Rarity { get; set; }

        [SearchableString]
        public string Type { get; set; }

        [SearchableString]
        public string Description { get; set; }

        [Sortable]
        [SearchableInteger]
        public int ElixirCost { get; set; }

        public string Targets { get; set; }

        public float HitSpeed { get; set; }

        public int Count { get; set; }

        public string Range { get; set; }

        public string Speed { get; set; }

        public float Radius { get; set; }

        public float DeployTime { get; set; }

        public int Lifetime { get; set; }

        public string DashRange { get; set; }

        public int ProjectileRange { get; set; }

        public Arena Arena { get; set; }

        public int Version { get; set; }

        public List<CardStatistics> CardStatistics { get; set; }
    }
}
